SynthSwap-Util/package/index.js

```diff
@@ -0,0 +1,1142 @@
+			},
+		],
+		name: 'uniswapV3SwapTo',
+		outputs: [
+			{
+				internalType: 'uint256',
+				name: 'returnAmount',
+				type: 'uint256',
+			},
+		],
+		stateMutability: 'payable',
+		type: 'function',
+	},
+	{
+		inputs: [
+			{
+				internalType: 'address payable',
+				name: 'recipient',
+				type: 'address',
+			},
+			{
+				internalType: 'contract IERC20',
+				name: 'srcToken',
+				type: 'address',
+			},
+			{
+				internalType: 'uint256',
+				name: 'amount',
+				type: 'uint256',
+			},
+			{
+				internalType: 'uint256',
+				name: 'minReturn',
+				type: 'uint256',
+			},
+			{
+				internalType: 'uint256[]',
+				name: 'pools',
+				type: 'uint256[]',
+			},
+			{
+				internalType: 'bytes',
+				name: 'permit',
+				type: 'bytes',
+			},
+		],
+		name: 'uniswapV3SwapToWithPermit',
+		outputs: [
+			{
+				internalType: 'uint256',
+				name: 'returnAmount',
+				type: 'uint256',
+			},
+		],
+		stateMutability: 'nonpayable',
+		type: 'function',
+	},
+	{
+		inputs: [
+			{
+				internalType: 'contract IERC20',
+				name: 'srcToken',
+				type: 'address',
+			},
+			{
+				internalType: 'uint256',
+				name: 'amount',
+				type: 'uint256',
+			},
+			{
+				internalType: 'uint256',
+				name: 'minReturn',
+				type: 'uint256',
+			},
+			{
+				internalType: 'bytes32[]',
+				name: 'pools',
+				type: 'bytes32[]',
+			},
+		],
+		name: 'unoswap',
+		outputs: [
+			{
+				internalType: 'uint256',
+				name: 'returnAmount',
+				type: 'uint256',
+			},
+		],
+		stateMutability: 'payable',
+		type: 'function',
+	},
+	{
+		inputs: [
+			{
+				internalType: 'contract IERC20',
+				name: 'srcToken',
+				type: 'address',
+			},
+			{
+				internalType: 'uint256',
+				name: 'amount',
+				type: 'uint256',
+			},
+			{
+				internalType: 'uint256',
+				name: 'minReturn',
+				type: 'uint256',
+			},
+			{
+				internalType: 'bytes32[]',
+				name: 'pools',
+				type: 'bytes32[]',
+			},
+			{
+				internalType: 'bytes',
+				name: 'permit',
+				type: 'bytes',
+			},
+		],
+		name: 'unoswapWithPermit',
+		outputs: [
+			{
+				internalType: 'uint256',
+				name: 'returnAmount',
+				type: 'uint256',
+			},
+		],
+		stateMutability: 'nonpayable',
+		type: 'function',
+	},
+	{
+		stateMutability: 'payable',
+		type: 'receive',
+	},
+];
+
+const swapFuncSelector = '7c025200000000000000000000000000';
+const uniswapV3SwapFuncSelector = 'e449022e000000000000000000000000';
+
+// query parameters:
+// see https://docs.1inch.io/docs/aggregation-protocol/api/swap-params#description-of-query-parameters
+
+// response parameters:
+// https://docs.1inch.io/docs/aggregation-protocol/api/swap-params#description-of-response-parameters
+
+// Swagger API: 
+// https://docs.1inch.io/docs/aggregation-protocol/api/swagger
+
+/// getFormatedSwapData returns data required by SynthSwap to execute swaps
+/// @param response: from /v4.0/10/swap
+/// @param synthwapAddress: address of deployed SynthSwap contract
+/// @returns Object: {functionSelector, data}
```

Comment:
Is this the 1inch `AggregationRouterV4.sol` function selector or `Synthswap` function selector?

Reply:
AggregationRouterV4.sol